core/classifiers/CodeClassifier/CodeClassifier.py

```diff
@@ -0,0 +1,710 @@
+        program: str. Input program for which tokens are to be generated.
+        VAR_TOKEN: str. Replace variable and method names with this string when
+            tokenizing data.
+
+    Returns:
+        list(str). A list containing tokens of input program.
+    """
+    tokenized_program = []
+    for token_id, token_name in get_tokens(program):
+        # Ignore all newline, comment and empty string tokens.
+        # token_id is token.N_TOKENS for empty line / newline token.
+        # token_id is 54 for comments.
+        if (token_id == token.N_TOKENS or token_id == COMMENT_TOKEN
+                or token_name.strip() == ''):
+            continue
+        elif token_id == token.NAME:
+            # If token_id is tokens.NAME then only add if it is a python
+            # keyword. Treat all variables and methods similar.
+            if token_name in keyword.kwlist:
+                tokenized_program.append(token_name)
+            else:
+                tokenized_program.append(VAR_TOKEN)
+        else:
+            tokenized_program.append(token_name)
+
+    return tokenized_program
+
+
+def map_tokens_to_ids(training_data, threshold):
+    """Generates a list of valid tokens and assigns a unique ID to each
+    token.
+
+    Args:
+        training_data: dict. A dictionary containing training_data. Structure of
+            training_data is as follows,
+                {
+                    ID: {
+                        'source': str. Source code of program.
+                        'class': int. Associated feedback class with program
+                    } for each program in training_data.
+                }
+        threshold: int. Ignore a token if it appears less than 'threshold' times
+            in dataset.
+
+    Returns:
+        dict. A dictionary containing tokens mapped to a unique ID.
+    """
+    # All unique tokens and number of time they occur in dataset. A token
+    # can be a keyword, an identifier, a constant, an operator.
+    vocabulary = {}
+
+    for pid in training_data:
+        program = training_data[pid]['source']
+        for token_id, token_name in get_tokens(program):
+            # Ignore all newline, comment and empty string tokens.
+            if (token_id == token.N_TOKENS or token_id == COMMENT_TOKEN
+                    or token_name.strip() == ''):
+                continue
+            # If token_id is tokens.NAME then only add if it is a python
+            # keyword.
+            elif token_id == token.NAME:
+                if token_name in keyword.kwlist:
+                    vocabulary[token_name] = vocabulary.get(token_name, 0) + 1
+            else:
+                vocabulary[token_name] = vocabulary.get(token_name, 0) + 1
+
+    # Consider only those tokens which occur for more than threshold times
+    # in entire dataset.
+    valid_tokens = [k for k, v in vocabulary.iteritems() if v > threshold]
+    token_to_id = dict(zip(valid_tokens, xrange(0, len(valid_tokens))))
```

Comment:
Yes, that sounds good to me. If you like, you can include a comment somewhere relevant with the canonical definitions that you want to use (if you think it would be helpful).